src/conf/settings.rs

```diff
@@ -101,40 +99,52 @@ impl Settings {
         if let Some(prefs_path) = bacon_prefs_path() {
```

Comment:
How about simplifying this to something which runs over an array of paths? e.g:

```rust
let paths = [
	bacon_prefs_path().filter(|p| p.exists()),
	config_path_from_env("BACON_PREFS"),
	context.workspace_config_path().filter(|p| p.exists()),
	context.workspace_dot_config_path().filter(|p| p.exists()),
	Some(context.package_config_path()).filter(|p| p.exists()),
	Some(context.package_dot_config_path()).filter(|p| p.exists()),
	config_path_from_env("BACON_CONFIG"),
];
for path in paths.flatten() {
	let prefs = Config::from_path(&path)?;
	settings.register_config_file(path);
    settings.apply_config(&prefs);
}
```

Perhaps make it a tuple if the debug statements are meaningful?

Reply:
I guess we can even factorize the `exist` test...

Reply:
I though a couple of those missing were perhaps because if the environment variable is set you might want to error or something like that, but either way works.